crazyflie/scripts/gui.py

```diff
@@ -0,0 +1,222 @@
+                # remove '/' and '/start_trajectory'
+                cfname = srv_name[1:-17]
+                if cfname != 'all':
+                    self.cfnames.append(cfname)
+
+        self.tf_buffer = Buffer()
+        self.tf_listener = TransformListener(self.tf_buffer, self)
+
+        self.cmd_vel_publisher = self.create_publisher(Twist, 'cmd_vel', 1)
+        self.sub_log = self.create_subscription(Log, 'rosout', self.on_rosout, rclpy.qos.QoSProfile(
+                        depth=1000,
+                        durability=rclpy.qos.QoSDurabilityPolicy.TRANSIENT_LOCAL))
+
+        self.logs = dict()
+        self.supervisor_labels = dict()
+        self.battery_labels = dict()
+        self.radio_labels = dict()
+        self.robotmodels = dict()
+        # set of robot names that had a status recently
+        self.watchdog = dict()
+
+        with Client.auto_index_client:
+
+            with ui.row().classes('items-stretch'):
+                with ui.card().classes('w-full h-full'):
+                    ui.label('Visualization').classes('text-2xl')
+                    with ui.scene(800, 400, on_click=self.on_vis_click) as scene:
+                        for name in self.cfnames:
+                            robot = scene.stl('/urdf/cf2_assembly.stl').scale(1.0).material('#ff0000').with_name(name)
+                            self.robotmodels[name] = robot
+                            self.watchdog[name] = time.time()
+                    scene.camera.x = 0
+                    scene.camera.y = -1
+                    scene.camera.z = 2
+                    scene.camera.look_at_x = 0  
+                    scene.camera.look_at_y = 0
+                    scene.camera.look_at_z = 0
+
+            with ui.row().classes('w-full h-lvh'):
+                with ui.tabs().classes('w-full') as tabs:
+                    self.tabs = []
+                    for name in ["all"] + self.cfnames:
+                        self.tabs.append(ui.tab(name))
+                with ui.tab_panels(tabs, value=self.tabs[0], on_change=self.on_tab_change).classes('w-full') as self.tabpanels:
+                    for name, tab in zip(["all"] + self.cfnames, self.tabs):
+                        with ui.tab_panel(tab):
+                            self.logs[name] = ui.log().classes('w-full h-96 no-wrap')
+                            self.supervisor_labels[name] = ui.label("")
+                            self.battery_labels[name] = ui.label("")
+                            self.radio_labels[name] = ui.label("")
+                            ui.button("Reboot", on_click=partial(self.on_reboot, name=name))
+
+            for name in self.cfnames:
+                self.create_subscription(Status, name + '/status', partial(self.on_status, name=name), 1)
+
+            # Call on_timer function
+            self.timer = self.create_timer(0.1, self.on_timer)
+
+    def send_speed(self, x: float, y: float) -> None:
```

Comment:
I don't see buttons for this (yet). Is this for a future functionality?

Reply:
This was simply dead code (copied from an official nicegui example).